### PR TITLE
png++ 0.2.9

### PR DESCRIPTION
**Pull request by hanazuki**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?
### Description

This patch upgrades png++ formula to the latest release 0.2.9. 0.2.9 requires patching to build with OS X clang.

### CODE REVIEW

File: Formula/png++.rb [hunk 1]
```diff
@@ -12,9 +11,74 @@ class Pngxx < Formula
     sha256 "f2e242ee428f191645418a9897eb2fd729408dd67d04e7af4cefc7dcb5715250" => :mavericks
   end
 
+  head do
+    url "svn://svn.savannah.nongnu.org/pngpp/trunk"
```
Comment (mikemcquaid) [inline Formula/png++.rb]: Can just make this `head "svn://svn.savannah.nongnu.org/pngpp/trunk"`

File: Formula/png++.rb [hunk 2]
```diff
@@ -12,9 +11,74 @@ class Pngxx < Formula
...
+  end
+
   depends_on "libpng"
 
+  patch :DATA
```
Comment (mikemcquaid) [inline Formula/png++.rb]: Can you please submit this patch to the upstream developers of this project and add a link to the upstream patch submission and explanation of why the patch is needed in a comment in the formula file. Thanks!

File: Formula/png++.rb [hunk 3]
```diff
@@ -12,9 +11,74 @@ class Pngxx < Formula
...
+        } catch(png::error &e) {
+          std::cerr << e.what() << std::endl;
+          throw;
+        }
+        return 0;
+      }
+    EOF
+
+    # 1x1 magenta image
+    (testpath/"1x1.png").write Base64.decode64 <<-EOF.undent
```
Comment (mikemcquaid) [inline Formula/png++.rb]: Can maybe just use `test_fixtures("test.png")` here instead.

Comment (hanazuki) [inline Formula/png++.rb]: Sure, I did not know about it. thanks

File: Formula/png++.rb (hunk 2 above)
Comment (hanazuki) [inline Formula/png++.rb]: It's submitted to the upstream and I have written some explanation in the patch header. I'll move it to here.

**Comment by hanazuki** [comment]
patch fixed according to the review by @mikemcquaid

File: Formula/png++.rb [hunk 4]
```diff
@@ -14,7 +14,61 @@ class Pngxx < Formula
 
   depends_on "libpng"
 
+  # Fix strerror_r usage for OS X clang
+  # patch from <https://savannah.nongnu.org/bugs/?46312>
+  patch :DATA
```
Comment (mikemcquaid) [inline Formula/png++.rb]: Would be good to submit an actual patch to them and a more detailed bug report. Thanks!

Comment (DomT4) [inline Formula/png++.rb]: Also would be nice to get the patch submitted to https://github.com/Homebrew/patches and use the commit reference from there in a `patch do` format. Inline patches behave fairly unpredictably at times, and people tend to make accidental whitespace changes to them on updates.

Comment (hanazuki) [inline Formula/png++.rb]: Hm, I'm not such motivated to do so. Sorry.

**Comment by hanazuki** [comment]
I've forgot to mention that the currently available version (0.2.5) has another bug and the formula needs fixing.

File: Formula/png++.rb (hunk 4 above)
Comment (mikemcquaid) [inline Formula/png++.rb]: We don't accept patches that have not been submitted upstream.

**Pull request closed without merging**